extensions/wikia/ImageReview/ImageReviewSpecialController.class.php

```diff
@@ -280,4 +199,112 @@ protected function parseImageData($data) {
 
 		return $images;
 	}
+
+	private function handlePostAction() {
+		$data = $this->wg->request->getValues();
+		if ( !empty($data) ) {
+			$images = $this->parseImageData($data);
+
+			if ( count($images) > 0 ) {
+				$this->helper->updateImageState( $images, $this->action );
+			}
+			$this->ts = null;
+		}
+	}
+
+	private function getOrderingMethod() {
+		if ($this->wg->user->isAllowed('imagereviewcontrols')) {
+			$order = (int)$this->getVal('sort', -1);
+			if ($order >= 0) {
+				$this->app->wg->User->setGlobalPreference('imageReviewSort', $order);
+				$this->app->wg->User->saveSettings();
+			}
+
+			$order = $this->app->wg->User->getGlobalPreference('imageReviewSort');
+			return $order;
+		} else {
+			$order = -1;
+			return $order;
+		}
+	}
+
+	private function getAction() {
```

Comment:
This name is a bit misleading. I would expect it to be some kind of accessor but it is more a parser. Fortunately it does not have side effects so it is not that bad. I can see that it follows convention in this class so it is up to you to change its name to something closer to what this function does.... `parseAction`?